### DISCUSSION TITLE
What is the code coverage in Rust?

### DISCUSSION THREAD

**Discussion by Rimantovas** [trustwallet/wallet-core#3902] [Q&A] [answered]
In a 2023 article you mentioned about migrating your existing C++ codebase to Rust. How much have you migrated so far? Is it best to use the C++ implementations or is Rust a viable solution?

**Comment by Milerius** [comment]
Hello @Rimantovas there is our current KPI in Rust:

- Progress transition of rust in % (for blockchains): 70.5%
- Progress transition of rust in % (for lines of code): 28.4%

Whenever u can we recommend the usage of Rust API, however even with 100% Rust, C++ Interface will still be available.

**Reply by Rimantovas** [reply]
Thanks for the quick reply. Can you suggest how to use the library in a rust project? I didn't find an exported crate to use. Also, couldn't find if you currently generate rust bindings from the existing C++ code or not.

**Reply by satoshiotomakan** [reply, marked as answer]
Hi @Rimantovas,
We do have plans to publish our crates to crates.io, but you can still use our Rust crates directly.
You only need to clone the whole https://github.com/trustwallet/wallet-core/ repository (no matter where to clone it).
Then you can import all required crates in Cargo.toml like this:
```toml
[dependencies]
tw_evm = { path = "path/to/wallet-core/rust/tw_evm" }
```

And then:
```rust
// main.rs
use tw_evm::address::Address;
use std::str::FromStr;

fn main() {
  let _ = Address::from_str("0xff00000000000000000000000000000000002694");
}
```

Please note that Rust crates have limited number of supported blockchains yet
> Also, couldn't find if you currently generate rust bindings from the existing C++ code or not.
